src/main/java/cz/xtf/openshift/OpenShiftWaiters.java

```diff
@@ -28,12 +28,44 @@
 	 * @return Waiter instance
 	 */
 	public Waiter hasBuildCompleted(String buildName) {
-		Supplier<String> supplier =  () -> openShiftUtil.getBuild(buildName).getStatus().getPhase();
+		Supplier<String> supplier = () -> openShiftUtil.getBuild(buildName).getStatus().getPhase();
 		String reason = "Waiting for completion of build " + buildName;
 
 		return new SupplierWaiter<>(supplier, "Complete"::equals, "Failed"::equals, TimeUnit.MINUTES, 10, reason).logPoint(Waiter.LogPoint.BOTH).interval(5_000);
 	}
 
+	/**
+	 * Creates waiter for latest build presence with default timeout
+	 * 5 seconds interval check and both logging points.
+	 *
+	 * @param buildConfigName name of buildConfig for which build to be waited upon
+	 * @return Waiter instance
+	 */
+	public Waiter isLatestBuildPresent(String buildConfigName) {
+		return isLatestBuildPresent(buildConfigName, null);
+	}
+
+	/**
+	 * Creates waiter for latest build presence with default timeout
+	 * 5 seconds interval check and both logging points.
+	 *
+	 * @param buildConfigName name of buildConfig for which build to be waited upon
+	 * @param buildNamespace namespace containing the buildConfig
+	 * @return Waiter instance
+	 */
+	public static Waiter isLatestBuildPresent(String buildConfigName, String buildNamespace) {
```

Comment:
OpenShiftWaiters shouldn't parameterize buildNamespace... it's namespace is parameterized by OpenShiftUtil. So if you want Waiter with different namespace you should do something like:

`OpenShiftUtil.master(buildNamespace).waiters().isLatestBuildPresent(buildConfigName);`

Reply:
Good catch, you are right. Was using these waiter methods in another helper class and I copypasted them here without thinking.